sdk/keyvault/cleanup/cleanup.cpp

```diff
@@ -0,0 +1,132 @@
+{
+  auto tenantId = std::getenv("AZURE_TENANT_ID");
+  auto clientId = std::getenv("AZURE_CLIENT_ID");
+  auto clientSecret = std::getenv("AZURE_CLIENT_SECRET");
+  auto credential
+      = std::make_shared<Azure::Identity::ClientSecretCredential>(tenantId, clientId, clientSecret);
+
+  SecretClient secretClient(std::getenv("AZURE_KEYVAULT_URL"), credential);
+  KeyClient keyClient(std::getenv("AZURE_KEYVAULT_URL"), credential);
+  CertificateClient certClient(std::getenv("AZURE_KEYVAULT_URL"), credential);
+
+  try
+  {
+    std::vector<DeleteCertificateOperation> certOps;
+    std::vector<DeleteKeyOperation> keyOps;
+    std::vector<DeleteSecretOperation> secretOps;
+
+    for (auto secrets = secretClient.GetPropertiesOfSecrets(); secrets.HasPage();
+         secrets.MoveToNextPage())
+    {
+      for (auto const& secret : secrets.Items)
+      {
+        try
+        {
+          secretOps.push_back(secretClient.StartDeleteSecret(secret.Name));
+          std::cout << "DeleteSecret" << secret.Name << std::endl;
+        }
+        catch (...)
+        {
+          std::cout << "fail to delete secret " << secret.Name;
+        }
+      }
+    }
+
+    for (auto certificates = certClient.GetPropertiesOfCertificates(); certificates.HasPage();
+         certificates.MoveToNextPage())
+    {
+      for (auto const& certificate : certificates.Items)
+      {
+        try
+        {
+          certOps.push_back(certClient.StartDeleteCertificate(certificate.Name));
+          std::cout << "Delete Certificate" << certificate.Name << std::endl;
+        }
+        catch (...)
+        {
+          std::cout << "fail to delete cert " << certificate.Name;
+        }
+      }
+    }
+
+    for (auto keys = keyClient.GetPropertiesOfKeys(); keys.HasPage(); keys.MoveToNextPage())
+    {
+      for (auto const& key : keys.Items)
+      {
+        try
+        {
+          keyOps.push_back(keyClient.StartDeleteKey(key.Name));
+          std::cout << "DeleteKey" << key.Name << std::endl;
+        }
+        catch (...)
+        {
+          std::cout << "fail to delete key " << key.Name;
+        }
+      }
+    }
+
+    for (auto op : certOps)
+    {
+      op.PollUntilDone(1s);
+      certClient.PurgeDeletedCertificate(op.Value().Name());
+      std::cout << "Purge cert " << op.Value().Name() << std::endl;
+    }
+
+    for (auto op : secretOps)
+    {
+      op.PollUntilDone(1s);
+      secretClient.PurgeDeletedSecret(op.Value().Name);
+      std::cout << "Purge secret " << op.Value().Name << std::endl;
+    }
+
+    for (auto op : keyOps)
+    {
+      op.PollUntilDone(1s);
+      keyClient.PurgeDeletedKey(op.Value().Name());
+      std::cout << "Purge Key " << op.Value().Name() << std::endl;
+    }
+  }
+  catch (Azure::Core::Credentials::AuthenticationException const& e)
+  {
+    std::cout << "Authentication Exception happened:" << std::endl << e.what() << std::endl;
+    return 1;
+  }
+  catch (Azure::Core::RequestFailedException const& e)
```

Comment:
Why not catch std::exception as well? Without that catch, the runtime library will abort, which can be quite problematic in our CI pipelines (especially on Windows).

Reply:
this is a utility executable that i use to cleanup the keyvault, i thought i would put it into the repo so others might use it, it is not build as part of the regular process